src/sql_query.ts

```diff
@@ -558,57 +558,61 @@ export default class SqlQuery {
         while (openMacros !== -1) {
-            let r = SqlQuery.betweenBraces(query.substring(openMacros+macros.length, query.length));
+            let r = SqlQuery.betweenBraces(query.substring(openMacros + macros.length, query.length));
             if (r.error.length > 0) {
                 throw {message: '$conditionalIn macros error: ' + r.error};
             }
             let arg = r.result;
-            // first parameters is an expression and require some complex parsing , so parse from the end where you know that the last parameters is a comma with a variable
-            let param1 = arg.substring(0,arg.lastIndexOf(',')).trim();
-            let param2 = arg.substring(arg.lastIndexOf(',')+1).trim();
-            // remove the $ from the variable 
+            // first parameters is an expression and require some complex parsing,
+            // so parse from the end where you know that the last parameters is a comma with a variable
+            let param1 = arg.substring(0, arg.lastIndexOf(',')).trim();
+            let param2 = arg.substring(arg.lastIndexOf(',') + 1).trim();
+            // remove the $ from the variable
             let varinparam = param2.substring(1);
             let done = 0;
-            //now find in the list of variable what is the value 
-	     for(var i=0;i<templateSrv.variables.length;i++){
-		var varG = templateSrv.variables[i];
-		if(varG.name===varinparam){
+            //now find in the list of variable what is the value
+            for (var i = 0; i < templateSrv.variables.length; i++) {
+                var varG = templateSrv.variables[i];
+                if (varG.name === varinparam) {
                     let closeMacros = openMacros + macros.length + r.result.length + 1;
                     done = 1;
-		    if(
-			    // for query variable when all is selected 
-			    // may be add another test on the all activation may be wise.
-			    (varG.type==='query' && ((varG.current.value.length==1 && varG.current.value[0]==='$__all') 
-                || (typeof(varG.current.value) === 'string' && varG.current.value === '$__all'))) ||
-			    // for textbox variable when nothing is entered 
-			    (varG.type==='textbox' && varG.current.value==='')){
-                        query = query.substring(0, openMacros)  + ' ' + query.substring(closeMacros, query.length);
-                      }else{
+
+                    const value = varG.current.value;
+
+                    if (
+                        // for query variable when all is selected
+                        // may be add another test on the all activation may be wise.
+                        (varG.type === 'query' && ((value.length == 1 && value[0] === '$__all')
+                            || (typeof value === 'string' && value === '$__all'))) ||
+                        // for textbox variable when nothing is entered
+                        (['textbox', 'custom'].includes(varG.type) && ['', undefined, null].includes(value))) {
```

Comment:
So this is the only significant change, everything else is just formatting

Reply:
formatting is https://github.com/Vertamedia/clickhouse-grafana/pull/176